xarray/core/dataarray.py

```diff
@@ -1519,12 +1519,68 @@ def interp(
 
         Examples
         --------
-        >>> da = xr.DataArray([1, 3], [("x", np.arange(2))])
-        >>> da.interp(x=0.5)
-        <xarray.DataArray ()>
-        array(2.)
+        da = xr.DataArray(
+            data=[
+                [0.1, 4.3, 2.1, 9.8],
+                [2.4, 7.9, 0.6, np.nan],
+                [6.9, np.nan, 5.3, 8.7],
+            ],
+            dims=("x", "y"),
+            coords={"x": [0, 1, 2], "y": [10, 12, 14, 16]},
+        )
+        >>> da
+        <xarray.DataArray (x: 3, y: 4)>
+        array([[0.1, 4.3, 2.1, 9.8],
+               [2.4, 7.9, 0.6, nan],
+               [6.9, nan, 5.3, 8.7]])
```

Comment:
Not worth changing for this one, but to the extent there's no specificity lost, having easy numbers so someone could work it out in their head makes it easy to understand what the function is doing _just_ with the examples